flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/sink/committer/FilesCommittable.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.flink.sink.committer;
+
+import java.io.Serializable;
+import org.apache.iceberg.io.WriteResult;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+
+public class FilesCommittable implements Serializable {
+  private final WriteResult committable;
+  private String jobId = "";
+  private long checkpointId;
+  private int subtaskId;
+
+  public FilesCommittable(WriteResult committable) {
```

Comment:
nit: is this constructor used?

Reply:
yes. 
https://github.com/apache/iceberg/blob/1b3feee3172bcde54670ca76db53d6c4507d83e8/flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/sink/writer/StreamWriter.java#L96-L100

Reply:
I also found it weird that we have two different constructors. if `prepareCommit` skip other fields in the `FilesCommittable`, will they ever be set? We only use the 2nd constructor in the deserializer path.